babel/dates.py

```diff
@@ -1282,6 +1284,16 @@ def format_timezone(self, char, num):
                 return get_timezone_name(self.value.tzinfo, width,
                                          uncommon=True, locale=self.locale)
             return get_timezone_location(self.value.tzinfo, locale=self.locale)
+        # Included additional elif condition to add support for 'X' in timezone format
+        elif char in "xX":
+            if get_timezone_gmt(self.value, width='short', locale=self.locale)=='+0000' and char == 'X':
+                return 'Z'
+            elif num == 1:
+                return get_timezone_gmt(self.value, width='short', locale=self.locale)[:-2]
```

Comment:
The reason I'd suggest to leave "O" formats for another PR is that [localizable GMT formats seem to be more complicated than Z/X](http://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology), since they need new data to be read from the CLDR XML files into our locale .dat files, and interpretation of that data.

Let's finish this first and then consider adding support for O? :)